fix remote repo teardown

When quickly starting a lot of (30) borg (1.0-maint branch) against same repository via ssh (to localhost), I frequently see this, but the majority of tasks work:

```
Remote: ssh_exchange_identification: Connection closed by remote host
Exception ignored in: <bound method RemoteRepository.__del__ of <RemoteRepository ssh://tw@localhost/home/tw/w/borg/repo>>
Traceback (most recent call last):
  File "/home/tw/w/borg/borg/remote.py", line 184, in __del__
    assert False, "cleanup happened in Repository.__del__"
AssertionError: cleanup happened in Repository.__del__
Connection closed by remote host. Is borg working on the server?
terminating with error status, rc 2
```

--- a/borg/remote.py
+++ b/borg/remote.py
@@ -170,12 +170,12 @@ class RemoteRepository:
         self.x_fds = [self.stdin_fd, self.stdout_fd, self.stderr_fd]
 
         try:
-            version = self.call('negotiate', RPC_PROTOCOL_VERSION)
-        except ConnectionClosed:
-            raise ConnectionClosedWithHint('Is borg working on the server?') from None
-        if version != RPC_PROTOCOL_VERSION:
-            raise Exception('Server insisted on using unsupported protocol version %d' % version)
-        try:
+            try:
+                version = self.call('negotiate', RPC_PROTOCOL_VERSION)
+            except ConnectionClosed:
+                raise ConnectionClosedWithHint('Is borg working on the server?') from None
+            if version != RPC_PROTOCOL_VERSION:
+                raise Exception('Server insisted on using unsupported protocol version %d' % version)
             # Because of protocol versions, only send append_only if necessary
             if append_only:
                 try:

